### PR TITLE
[Core] Added TextTransform property

### LINKED ISSUE #3667: [Enhancement] Add text-transforms to Label
## Summary
Our design-people came up with a design that includes headers that are all uppercase. Right now, we use a converter for this or included it in a markup extension for values that we localize. I think it would be great to have a couple of text transforms (a la CSS) in Forms available by default.

## API Changes

It makes the most sense to implement this on the `Label` control. A new property would be introduced taking in a new enum. You can see them underneath:

- `Label.TextTransform` value of `TextTransform` enum
- `TextTransform` enum: `None`, `Uppercase`, `Lowercase`, `Title`, `None` will be the default to be backwards compatible.

## Intended Use Case
As mentioned already in the summary; you might encouter use-cases where you want to implement text with a certain casing. While this is solvable by entering a string uppercase, writing your own extension, etc. In my opinion this is purely a styling thing and belongs to a property on the control so you don't have to think about this anymore.

### PR DESCRIPTION
**Pull request by paymicro**
### Description of Change ###

Added TextTransform property to the Label

### Issues Resolved ### 

- fixes #3667

### API Changes ###

Added:
 - TextTransform InputView.TextTransform { get; set; } //Bindable Property

```cs
Internals.TextTransform.Uppercase
Internals.TextTransform.Lowercase
Internals.TextTransform.None
```

### Platforms Affected ### 

- Core (all platforms)

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

- run Issue 3667
- click on button "Change TextTransform"
- check result

### PR Checklist ###

- [x] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Core/Label.cs [hunk 1]
```diff
@@ -127,10 +164,43 @@ public LineBreakMode LineBreakMode
...
+				{
+					sb.Append(char.ToUpperInvariant(c));
+					beginningOfWord = false;
+					continue;
+				}
+				sb.Append(char.ToLowerInvariant(c));
+			}
+			return sb.ToString();
+#else
+			return System.Globalization.CultureInfo.CurrentCulture.TextInfo.ToTitleCase(input.ToLower());
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Label.cs]: TitleCase is more complex than this, some words shouldn't be capitalized.

Comment (StephaneDelcroix) [reply Xamarin.Forms.Core/Label.cs]: TitleCase is valid for english texts. Different culture capitalize their texts differently

**Comment by jfversluis** [comment]
How are we doing on this? Still feel it would be a valuable addition. @paymicro can you still do it? Else I might take a look :)

**Comment by paymicro** [comment]
@rmarinho on Android text changes on one way... It may be wrong to lose original text.

**Comment by rmarinho** [comment]
@paymicro can you rebase please?

**Comment by rmarinho** [comment]
@paymicro I wonder if we could while we are doing this add a `virtual` method called UpdateFormsText on the Renderers or something, so users could more easily override this behaviour if they want since we are messing with that part of the code.

**Comment by jsuarezruiz** [comment]
@paymicro Can you rebase please?.

**Comment by jfversluis** [comment]
@jsuarezruiz @paymicro I have taken the liberty to do some work on this. Rebased and made it work together with the character spacing stuff!

Please double-check to see if all went well

**Comment by samhouts** [comment]
@rmarinho Issue3667.Issue3667Tests is not passing.

**Comment by davidortinau** [comment]
What does this do? I see @StephaneDelcroix suggesting not to handle TitleCase, but the original description includes no indication of what transforms are supported.

It's the casing transform I'm interesting in, on a button. #5516

**Comment by jsuarezruiz** [comment]
droid3690

**Comment by rmarinho** [comment]
@jsuarezruiz i fixed that issue on Android , can you double check ?! 
THanks

**Comment by Mikilll94** [comment]
In which version of XF this feature is available?

**Comment by jfversluis** [comment]
@Mikilll94 this was merged into master so that would be 4.8

**Comment by davidortinau** [comment]
This doesn't work when using FormattedString or on Span. Tested on Android API29

**Pull request merged**